### PR TITLE
Handle PATCH requests to /api/articles/:article_id

### PR DESCRIPTION
**Pull request by schaxmann**
Queries:
Is it really bad to use a nested .then block (app.test.js:85)?
Am I miles off with 400 vs 422 responses in my error handling?

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -64,5 +64,77 @@ describe("news api", () => {
           });
       });
     });
+    describe("PATCH", () => {
+      test("201: returns an updated article object, containing author, title, article_id, body, topic, created_at & incremented votes properties", () => {
+        const articleUpdateObj = { inc_votes: 25 };
```
Comment (Poonam-raj) [inline __tests__/app.test.js:69]: a patch would be status 200 instead of 201 Created

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Some changes needed in the testing so I'm not approving but code is looking strong.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -64,5 +64,77 @@ describe("news api", () => {
...
+          .send(invalidUpdate)
+          .expect(422)
+          .then(({ body }) => {
+            const UnprocessableEnt = body.msg;
+            expect(UnprocessableEnt).toBe(
+              "Unprocessable Entity. Request must include an inc_votes key with a number value"
+            );
+          });
+      });
+    });
```
[the remark this answers is not in the document]
Comment (schaxmann) [reply __tests__/app.test.js:138]: Thank you for all of your advice Poonam. Have implemented outlined changes (I hope!).

**Review by Rayhaan2k9 — APPROVED** [review]
_Now, this is a story all about how
My life got flipped-turned upside down
And I'd like to take a minute
Just sit right there
I'll tell you how I became the prince of a town called Bel-Air
In West Philadelphia born and raised
On the playground was where I spent most of my days
Chillin' out, maxin', relaxin', all cool
And all shootin' some b-ball outside of the school
When a couple of guys who were up to no good
Started making trouble in my neighborhood
I got in one little fight and my mom got scared
She said, "You're movin' with your auntie and uncle in Bel-Air"_

All good! 😄

**Pull request merged**